### DISCUSSION TITLE
Startup sessions, splits and window sizing issues

### DISCUSSION THREAD

**Discussion by pm5k** [kovidgoyal/kitty#3652] [Q&A]
Hi, I am having trouble figuring out what is the standard (accepted) way to setup a startup session which contains properly sized window arrangements. I cannot find anything in the docs to show how would one achieve two distinct Kitty os windows side-by-side, but the first one is sized to take over 30% of the available screen space and the other - 70%, for example.

SO, what I have so far is this:

```bash
layout splits:axis=horizontal
launch --title=Neofetch --location=hsplit zsh -c "sleep 0.5; neofetch; zsh"
launch --title=Sys --location=hsplit btm
new_os_window
layout splits
launch --title=Projects --location=hsplit --cwd=~/Projects zsh
focus
launch --title=Generic --location=hsplit zsh
```

Which results in two os windows (from what I can tell, this is what you call them in the docs?), that share 50% of the space (each) on the desktop. I wish to make the first one only take up X%, while the remaining one should take the rest (but I do not mind specifying this manually).

Is this possible? I have resorted to this, because I frankly could not, for the life of me, figure out how to do the following in a single OS window using layouts (again, docs hint at splits being the right choice, but the way splits are auto-portioned made it impossible to achieve what I wanted):

```
+----------OS_Window----------+
|          |                  |
| neofetch |      zsh         |
|          |                  |
|          |------------------|
|          |                  |
|----------|      zsh         |
| btm      |                  |
+----------+------------------+
```

I tried making this layout using a combo of `hsplit` and `vsplit` and I just could not achieve it with any combo of hsplit/vsplit on the four launch instructions I had defined in the session file.

So I guess my question is two-fold.. Can the above ASCII table be expressed as a single os_window layout using splits? If so - how do I achieve that -- since I have not found a way to portion those splits in such a way that they respect that sort of layout in a single OS window. And if this is impossible -- how do I achieve a startup session with two os windows, one of which only takes up X% and the other takes up Y% horizontal space (whichever is easiest, I suppose, but I would prefer a single window with splits).

I'm a little reluctant to write this, but frankly I have spent way too long trying to figure this out on my own and would love some guidance as the docs were of no help here.

Thanks in advance.

**Comment by kovidgoyal** [comment]
There is no way to open splits at particular sizes using launch. However, what you can do instead is create the splits, then resize them using remote control, https://sw.kovidgoyal.net/kitty/remote-control.html  Someday I might add a way to pass layout dependent options like biases for sizing to launch.

**Comment by salmankhilji** [comment]
See #3620.
